SLHCUpgradeSimulations/Phase2TrackerDTCCablingMap/plugins/DTCCablingMapProducer.cc

```diff
@@ -0,0 +1,323 @@
+		unsigned                        csvFormat_ielinkid_                     ;
+		cond::Time_t                    iovBeginTime_                           ;
+		TrackerDetToDTCELinkCablingMap* pCablingMap_                            ;
+		std::string                     record_                                 ;
+};
+
+
+void DTCCablingMapProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions)
+{
+	edm::ParameterSetDescription desc;
+	desc.setComment("Stores a TrackerDetToDTCELinkCablingMap object into the database from a CSV file.");
+	desc.add<bool>("generate_fake_valid_gbtlink_and_elinkid", false);
+	desc.add<int>("verbosity", 0);
+	desc.add<unsigned>("csvFormat_idetid"           ,  0);
+	desc.add<unsigned>("csvFormat_ncolumns"         , 15);
+	desc.add<unsigned>("csvFormat_idtcid"           , 10);
+	desc.add<unsigned>("csvFormat_igbtlinkid"       , 11);
+	desc.add<unsigned>("csvFormat_ielinkid"         , 12);
+  desc.add<long long unsigned int>("iovBeginTime" ,  1);
+  desc.add<std::string>("record","TrackerDTCCablingMapRcd");
+	desc.add<std::vector<std::string>>("modulesToDTCCablingCSVFileNames",std::vector<std::string>());
+	descriptions.addDefault(desc);
+}
+
+DTCCablingMapProducer::DTCCablingMapProducer(const edm::ParameterSet& iConfig):
+	generate_fake_valid_gbtlink_and_elinkid_ (iConfig.getParameter<bool>("generate_fake_valid_gbtlink_and_elinkid")),
+	verbosity_                               (iConfig.getParameter<int>("verbosity")),
+	csvFormat_ncolumns_                      (iConfig.getParameter<unsigned>("csvFormat_ncolumns")),
+	csvFormat_idetid_                        (iConfig.getParameter<unsigned>("csvFormat_idetid"  )),
+	csvFormat_idtcid_                        (iConfig.getParameter<unsigned>("csvFormat_idtcid"  )),
+	iovBeginTime_                            (iConfig.getParameter<long long unsigned int>("iovBeginTime")),
+	pCablingMap_                             (new TrackerDetToDTCELinkCablingMap()),
+	record_                                  (iConfig.getParameter<std::string>("record"))
+{
+	LoadModulesToDTCCablingMapFromCSV(iConfig.getParameter<std::vector<std::string>>("modulesToDTCCablingCSVFileNames"));
+}
+
+void DTCCablingMapProducer::beginJob()
+{
+	
+}
+
+void DTCCablingMapProducer::LoadModulesToDTCCablingMapFromCSV(std::vector<std::string> const& modulesToDTCCablingCSVFileNames)
+{
+// 	std::vector<std::string>
+// 	edm::FileInPath
+// 	iConfig.getParameter<edm::FileInPath>("inputCablingFileName").fullPath().c_str()
+	
+	using namespace std;
+	
+	for (std::string const& csvFileName : modulesToDTCCablingCSVFileNames)
+	{
+		edm::FileInPath csvFilePath(csvFileName);
+		
+		ifstream csvFile;
+		csvFile.open(csvFilePath.fullPath().c_str());
+		
+		if (csvFile.is_open())
+		{
+			string csvLine;
+			
+			unsigned lineNumber = 0;
+			
+			while (std::getline(csvFile, csvLine))
+			{
+				if (verbosity_ >= 1)
+				{
+					cout << "Reading CSV file line: " << ++lineNumber << ": \"" << csvLine << "\"" << endl;
```

Comment:
here and below `cout` is discouraged as it is not thread-safe. You can use Message Logger  (`edm:LogInfo` etc. instead)

Reply:
Ok, I'll change it